test/unit/domain_test.rb

```diff
@@ -20,6 +20,14 @@ def setup
     assert !other_domain.save
   end
 
+  test "should have a unique case-insensitive name" do
+    other_domain = Domain.new(:name => @domain.name.upcase)
+    refute other_domain.save
+    assert_raises ActiveRecord::RecordNotUnique, ActiveRecord::StatementInvalid do
```

Comment:
@dLobatog ping, trying to revive this old PR. Could I get you input about the previous comment?